internal/tofu/node_provider.go

```diff
@@ -41,7 +41,11 @@ func (n *NodeApplyableProvider) Execute(ctx EvalContext, op walkOperation) (diag
 	switch op {
 	case walkValidate:
 		log.Printf("[TRACE] NodeApplyableProvider: validating configuration for %s", n.Addr)
-		return diags.Append(n.ValidateProvider(ctx, provider))
+		diags = diags.Append(n.ValidateProvider(ctx, provider))
+		if diags.HasErrors() {
+			return diags
+		}
+		fallthrough
```

Comment:
How about, instead of fallthrough, moving walkValidate to the below case, and then have an `if` at the beginning. Fallthrough is easy to break in unexpected ways when moving code around.